src/EditorFeatures/Core/Implementation/CommentSelection/AbstractCommentSelectionBase.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                document.Project.Solution.Workspace.ApplyTextChanges(document.Id, edits.TextChanges.Distinct(), CancellationToken.None);
+                transaction.Complete();
+            }
+
+            // Convert the tracking spans into snapshot spans for formatting and selection.
+            var trackingSnapshotSpans = trackingSpans.Select(s => CreateSnapshotSpan(subjectBuffer.CurrentSnapshot, s.trackingSpan, s.originalSpan));
+
+            if (trackingSnapshotSpans.Any())
+            {
+                if (edits.ResultOperation == Operation.Uncomment)
+                {
+                    // Format the document only during uncomment operations.  Use second transaction so it can be undone.
+                    using (var transaction = new CaretPreservingEditTransaction(title, textView, _undoHistoryRegistry, _editorOperationsFactoryService))
+                    {
+                        var formattedDocument = Format(service, subjectBuffer.CurrentSnapshot, trackingSnapshotSpans, CancellationToken.None);
+                        formattedDocument?.Project.Solution.Workspace.ApplyDocumentChanges(formattedDocument, CancellationToken.None);
+                        transaction.Complete();
+                    }
+                }
+
+                var spansToSelect = trackingSnapshotSpans.Select(s => new Selection(s));
+                // Set the multi selection with the last selection as the primary after edits have been applied.
+                textView.GetMultiSelectionBroker().SetSelectionRange(spansToSelect, spansToSelect.Last());
+            }
+        }
+
+        /// <summary>
+        /// Creates a tracking span for the operation.
+        /// Internal for tests.
+        /// </summary>
+        internal static ITrackingSpan CreateTrackingSpan(Operation operation, ITextSnapshot snapshot, TextSpan textSpan)
+        {
+            // If a comment is being added, the tracking span must include changes at the edge.
+            var spanTrackingMode = operation == Operation.Comment
+                ? SpanTrackingMode.EdgeInclusive
+                : SpanTrackingMode.EdgeExclusive;
+            return snapshot.CreateTrackingSpan(Span.FromBounds(textSpan.Start, textSpan.End), spanTrackingMode);
+        }
+
+        /// <summary>
+        /// Retrieves the snapshot span from a post edited tracking span.
+        /// Additionally applies any extra modifications to the tracking span post edit.
+        /// Internal for tests.
+        /// </summary>
+        private static SnapshotSpan CreateSnapshotSpan(ITextSnapshot snapshot, ITrackingSpan trackingSpan, CommentTrackingSpan originalSpan)
+        {
+            var snapshotSpan = trackingSpan.GetSpan(snapshot);
+            if (originalSpan.HasPostApplyChanges())
+            {
+                var updatedStart = snapshotSpan.Start.Position + originalSpan.AmountToAddToTrackingSpanStart;
```

Comment:
what does it actually t rying to do? just curious.

Reply:
I add both *//* when the selection starts or ends inside a block comment.  The selection should include only one of the comment markers, so I need to modify the tracking span after it has been applied to only include one.  The span tracking modes will only allow me to include all or none of the markers (instead of just 1).